index.bs

```diff
@@ -769,63 +809,84 @@ spec:css-syntax-3;
 
-    6.  Let |sameOriginWithAncestors| be `true` if |settings| is [=same-origin with its
+    1.  Let |p| be [=a new promise=].
+
+    1.  Let |origin| be the [=current settings object=]'s [=environment settings object/origin=].
+
+    1.  Let |sameOriginWithAncestors| be `true` if |settings| is [=same-origin with its
         ancestors=], and `false` otherwise.
 
-    7.  Run the following steps [=in parallel=]:
+    1.  Run the following steps [=in parallel=]:
 
         1.  Let |credentials| be the result of <a abstract-op lt="collect local">collecting
             `Credential`s from the credential store</a>, given |origin|, |options|, and
             |sameOriginWithAncestors|.
 
-        2.  If |credentials| is an [=exception=], [=reject=] |p| with |credentials|.
+        1.  If |credentials| is an [=exception=], [=reject=] |p| with |credentials|.
 
-        3.  If all of the following statements are true, resolve |p| with |credentials|[0] and
+        1.  If all of the following statements are true, resolve |p| with |credentials|[0] and
             skip the remaining steps:
 
             1.  |credentials|' [=set/size=] is 1
 
-            2.  |origin| does not [=origin/requires user mediation|require user mediation=]
+            1.  |origin| does not [=origin/requires user mediation|require user mediation=]
 
-            3.  |options| is <a>matchable <i lang="la">a priori</i></a>.
+            1.  |options| is <a>matchable <i lang="la">a priori</i></a>.
 
-            4.  |options|.{{CredentialRequestOptions/mediation}} is not
+            1.  |options|.{{CredentialRequestOptions/mediation}} is not
                 "{{CredentialMediationRequirement/required}}".
 
+            1.  |options|.{{CredentialRequestOptions/mediation}} is not
+                "{{CredentialMediationRequirement/conditional}}".
+
             ISSUE: This might be the wrong model. It would be nice to support a site that wished
             to accept either username/passwords or webauthn-style credentials without forcing
             a chooser for those users who use the former, and who wish to remain signed in.
 
-        4.  If |options|' {{CredentialRequestOptions/mediation}} is
+        1.  If |options|' {{CredentialRequestOptions/mediation}} is
             "{{CredentialMediationRequirement/silent}}", [=resolve=] |p| with `null`, and skip
             the remaining steps.
 
-        5.  Let |choice| be the result of <a abstract-op lt="ask to choose">asking the user to
+        1.  Let |choice| be the result of <a abstract-op lt="ask to choose">asking the user to
             choose a `Credential`</a>, given |options| and |credentials|.
 
-        6.  If |choice| is `null` or a {{Credential}}, [=resolve=] |p| with |choice| and skip the
-            remaining steps.
+        1.  If |choice| is not `null` and is not a {{Credential}}:
+
+            1.  Assert: |choice| is an [=interface object=].
```

Comment:
How about:

```markdown
        1.  Let |result| be the result of <a abstract-op lt="ask to choose">asking the user to
            choose a `Credential`</a>, given |options| and |credentials|.

        1.  If |result| is an [=interface object=]:

            1.  Set |result| to the result of executing |result|'s
                {{[[DiscoverFromExternalSource]](origin, options, sameOriginWithAncestors)}},
                given |origin|, |options|, and |sameOriginWithAncestors|.

        1.  Assert: |result| is `null`, a {{Credential}}, or an [=exception=].
```

This asserts a slightly different thing, so you could use <code>Assert: |result| is `null`, a {{Credential}}, or an [=interface object=].</code> above the `if` instead.

Reply:
This is cleaner, and I like asserting over the result of [[DiscoverFromExternalSource]] as well, thanks.